src/arkode/arkode_io.c

```diff
@@ -885,6 +885,29 @@ int ARKodeSetUserData(void* arkode_mem, void* user_data)
   return (ARK_SUCCESS);
 }
 
+/*---------------------------------------------------------------
+  ARKodeSetOwnUserData:
+
+  Set whether the ARKodeMem structure owns the user data pointer.
+  If set to SUNTRUE, ARKode will free the user data pointer when
+  ARKodeFree is called.
+  ---------------------------------------------------------------*/
+int ARKodeSetOwnUserData(void* arkode_mem, sunbooleantype own_user_data)
```

Comment:
Prefix internal function with `ark`
```suggestion
int arkSetOwnUserData(void* arkode_mem, sunbooleantype own_user_data)
```